fate/src/main/java/org/apache/accumulo/fate/util/LoggingRunnable.java

```diff
@@ -34,20 +34,37 @@ public void run() {
     try {
       runnable.run();
     } catch (Throwable t) {
+      boolean runnableError = (t instanceof Error);
       try {
         log.error("Thread \"{}\" died {}", Thread.currentThread().getName(), t.getMessage(), t);
       } catch (Throwable t2) {
-        // maybe the logging system is screwed up OR there is a bug in the exception, like
-        // t.getMessage() throws a NPE
-        System.err.println(
-            "ERROR " + new Date() + " Failed to log message about thread death " + t2.getMessage());
-        t2.printStackTrace();
-
-        // try to print original exception
-        System.err
-            .println("ERROR " + new Date() + " Exception that failed to log : " + t.getMessage());
-        t.printStackTrace();
+        boolean logError = (t2 instanceof Error);
+        try {
+          // maybe the logging system is screwed up OR there is a bug in the exception, like
+          // t.getMessage() throws a NPE
+          System.err.println("ERROR " + new Date() + " Failed to log message about thread death "
+              + t2.getMessage());
+          t2.printStackTrace();
+
+          // try to print original exception
+          System.err
+              .println("ERROR " + new Date() + " Exception that failed to log : " + t.getMessage());
+          t.printStackTrace();
+        } catch (Throwable t3) {
+          // If printing to System.err didn't work then don't try to log that exception but do
+          // re-throw it if
+          // it's the most serious failure we've seen so far.
+          boolean printError = (t3 instanceof Error);
+          if (printError && !(logError || runnableError))
```

Comment:
This is far more readable:

```suggestion
          if (printError && !logError && !runnableError)
```

Reply:
Sure. I'll roll that into the next commit.